Makefile

```diff
@@ -29,6 +29,22 @@ IGNORE_COVERAGE += github.com/veraison/services/plugin/test
 # There is protobuf-generated stuff here, which skews coverage.
 IGNORE_COVERAGE += github.com/veraison/services/handler
 
+# Go 1.22 started reporting subpackages without any tests as having 0.0%
```

Comment:
Track what? It's a change in behavior of Go going forward. The way to "fix" this is to filter out subpackages with no tests ourselves, since Go will no longer ingore them. This change is already doing that. I don't think there is anything else we need to do for this?

Reply:
We need to increase coverage rather than hide the signal coming from the toolchain.

Reply:
I'm not sure that would be useful.

If you look at the subpackages added,  all of that code is already being tested. Extensively.  Either as part of integration tests (for the service exectuables) or form other subpackages (the scheme/common stuff).

This a problem with how Go deails with coverage -- it considers each subpackage indivdually rather than the code base as a whole.

Yes, we _could_ write unit tests in every subpackage, however they contain code that isn't exercised easily in isolation, so that would require a lot of mocking and harnessing. It's going to be significant effort to add that test infartructure to begin with, and then a whole lot more to maintain it going forward (this pull request being a case in point). And, at the end of the day, we're not actually going to be testing anything that's not already being tested.

It's not worth it.

Reply:
I am pretty sure not all the sub-packages are worth adding tests for.

What I am saying is we should have a critical look at the list of the currently excluded packages and consciously decide whether they are worth some coverage or not.

Reply:
Ok, that makes sense.

Reply:
https://github.com/veraison/services/issues/252